src/components/header/header.jsx

```diff
@@ -0,0 +1,67 @@
+import React, {useCallback} from 'react';
+import {propTypes} from './header.props';
+
+const Header = (props) => {
+  const {
+    isAuthorized,
+    userEmail,
+    onLoginClick,
+    onProfileClick = ()=>{},
+  } = props;
+  const handleLoginClick = useCallback((evt) => {
+    evt.preventDefault();
+    onLoginClick();
+  }, []);
```

Comment:
твой callback должен меняться если пришел другой  `onLoginClick` иначе ты будешь вызывать прошлую функцию. Знаешь что такое мемоизация?
```suggestion
  }, [onLoginClick]);
```